### PR TITLE
HCL2\Terraform v0.12 conversion

### PR DESCRIPTION
**Pull request by twistedgrim**
##### Corresponding Issue(s):
 - PRs should have a corresponding issue. If no issue exists, provide details in the **Reason for Change(s)** section
[MPCSUPENG-869](https://jira.rax.io/browse/MPCSUPENG-869)
##### Summary of change(s):
- Module updated from HCL1 syntax to HCL2
- Module configured with a minimum terraform version of 0.12.0 and minimum AWS provider version of 2.2.
- Formatting and code changes due to HCL2 conversion and to meet Rackspace Terraform Style Guide requirements
##### Reason for Change(s):
Terraform 0.12 support

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:
No, none of the changes should result in the replacement of resources, however several resources may be updated during the process, such as tags, SSM Document spacing, etc.
Test infrastructure was updated to severely reduce the number of resources created.

##### If input variables or output variables have changed or has been added, have you updated the README?
Readme should be up to date.

##### Do examples need to be updated based on changes?
Examples were converted during the process and updated to the expected release number.

##### Note to the PR requester about Closing PR's
Please message the person that opened the issue when auto closing it on slack, as well as any other stake holders of deep interest. Only close the issue if you believe that the issue is fully resolved with this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,89 +1,115 @@
...
+### Module variables
+
+The following module variables were updated to better meet current Rackspace style guides:
+
+- `security_group_list` -> `security_groups`  
+- `resource_name` -> `name`
+
+The following variables are no longer neccessary and were removed
+
+- `additional_ssm_bootstrap_step_count
```
Comment (johnctitus) [inline README.md]: Missing the closing backtick on this line

File: examples/main.tf [hunk 2]
```diff
@@ -79,6 +80,7 @@ module "ec2_ar" {
         "timeoutSeconds": 300
       }
 EOF
+
```
Comment (johnctitus) [inline examples/main.tf]: I would suggest converting this older format\field to the newer one `ssm_bootstrap_list`

File: examples/main.tf [hunk 3]
```diff
@@ -95,21 +97,22 @@ EOF
...
   cw_cpu_high_evaluations             = "15"
+  cw_cpu_high_operator                = "GreaterThanThreshold"
   cw_cpu_high_period                  = "60"
+  cw_cpu_high_threshold               = "90"
+  disable_api_termination             = "False"
+  eip_allocation_id_count             = "3"
+  eip_allocation_id_list              = [aws_eip.my_eips.*.id]
+  notification_topic                  = ""
+  private_ip_address                  = ["10.0.1.131", "10.0.1.132", "10.0.1.133"]
+  t2_unlimited_mode                   = "standard"
```
Comment (johnctitus) [inline examples/main.tf]: We should rename the `additional_tags` field below this line to just `tags`

File: main.tf [hunk 4]
```diff
@@ -27,13 +26,178 @@
...
+ * ### Module variables
+ *
+ * The following module variables were updated to better meet current Rackspace style guides:
+ *
+ * - `security_group_list` -> `security_groups`
+ * - `resource_name` -> `name`
+ *
+ * The following variables are no longer neccessary and were removed
+ *
+ * - `additional_ssm_bootstrap_step_count
```
Comment (johnctitus) [inline main.tf]: Source for the missing backtick mentioned earlier.

File: text/managed_ssm_steps.json [hunk 5]
```diff
@@ -1,20 +1,19 @@
     {
```
Comment (johnctitus) [inline text/managed_ssm_steps.json]: I think this file is no longer needed.

File: tests/test1/main.tf [hunk 6]
```diff
@@ -12,94 +12,59 @@ resource "random_string" "res_name" {
 }
 
 module "vpc" {
-  source = "git@github.com:rackspace-infrastructure-automation/aws-terraform-vpc_basenetwork?ref=master"
+  source = "git@github.com:rackspace-infrastructure-automation/aws-terraform-vpc_basenetwork?ref=v0.12.0"
```
Comment (johnctitus) [inline tests/test1/main.tf]: We probably should not pin the tests to a particular release, just to make sure we continue to test vs the latest features

File: main.tf [hunk 7]
```diff
@@ -27,13 +26,178 @@
...
+ *
+ * Additionally, a new variable `tags` was added to replace the functionality of the `additional_tags` variable.  `tags` allows setting tags on all resources.
```
Comment (johnctitus) [inline main.tf]: Should remove this line and just add additional_tags -> tags to the list of renamed variables.

File: README.md (hunk 1 above)
Comment (twistedgrim) [reply README.md]: fixed

File: examples/main.tf (hunk 2 above)
Comment (twistedgrim) [reply examples/main.tf]: changed to new style

File: examples/main.tf (hunk 3 above)
Comment (twistedgrim) [reply examples/main.tf]: cleaned up

File: text/managed_ssm_steps.json (hunk 5 above)
Comment (twistedgrim) [reply text/managed_ssm_steps.json]: gone

File: main.tf (hunk 4 above)
Comment (twistedgrim) [reply main.tf]: fixed in main

File: tests/test1/main.tf (hunk 6 above)
Comment (twistedgrim) [reply tests/test1/main.tf]: back to master. I think I just had this for testing.

File: main.tf (hunk 7 above)
Comment (twistedgrim) [reply main.tf]: gone

**Pull request merged**